### ISSUE TITLE
disallowed-methods not working

### ISSUE THREAD

**Issue by xxchan** [rust-lang/rust-clippy#8577]
### Summary

```toml
# in clippy.toml
disallowed-methods = [
     "std::boxed::Box::new"
]
```
```rust
// in main.rs

fn main() {
    let _box = Box::new(3); 
}
```

no warning
```bash
❯ cargo clean; cargo clippy 
    Checking nono v0.1.0 (D:\tmp\nono)
    Finished dev [unoptimized + debuginfo] target(s) in 0.25s
```

### Lint Name

disallowed-methods

### Version

```text
rustc 1.58.1 (db9d1b20b 2022-01-20)
binary: rustc
commit-hash: db9d1b20bba1968c1ec1fc49616d4742c1725b4b
commit-date: 2022-01-20
host: x86_64-pc-windows-msvc
release: 1.58.1
LLVM version: 13.0.0

also for 

rustc 1.61.0-nightly (1eb72580d 2022-03-08)
binary: rustc
commit-hash: 1eb72580d076935a3e590deb6e5813a5aef3eca4
commit-date: 2022-03-08
host: x86_64-unknown-linux-gnu
release: 1.61.0-nightly
LLVM version: 14.0.0
```

**Comment by Jarcho** [comment]
The proper path for `Box` is `alloc::boxed::Box`. `std` just re-exports it. It would be nice if it worked, but I don't think we have easy access to re-exported names from clippy.

**Comment by xxchan** [comment]
I tried v1.49.0, which is the first version supporting disallowed_methods. But it also didn't work.🤨

**Comment by Jarcho** [comment]
Just tested on `1.59.0`. It works with both `std::boxed::Box::new` and `alloc::boxed::Box::new`. I had to explicitly enable the lint though.

**Comment by xxchan** [comment]
Oh indeed. I misunderstood the clippy config file... Thank you for your hint.

Now it works fine for `Box::new`.

But the method I actually wanted to disallow is `fixedbitset::FixedBitSet::from_iter`. It seems that the trait method cannot be disallowed?

**Comment by Jarcho** [comment]
The lint is /supposed/ to be enabled by default. I don't know what's happening there.

For trait methods it might only work if you name the method on the trait itself, rather than on the type that implements it. Less than useful in this case, unfortunately.

**Comment by xxchan** [comment]
I found actually it works on v1.61.0-nightly. It may because https://github.com/rust-lang/rust-clippy/pull/8261 is quite recent.

Anyway thanks for your help @Jarcho!

**Comment by Jarcho** [comment]
Just opened #8581 about disallowing methods on trait impls.
